collect_app/src/main/java/org/odk/collect/android/activities/GeoPointMapActivity.java

```diff
@@ -131,7 +131,7 @@ public void onCreate(Bundle savedInstanceState) {
         isMapReady = false;
         ((SupportMapFragment) getSupportFragmentManager().findFragmentById(R.id.map)).getMapAsync(googleMap -> {
             setupMap(googleMap);
-            locationClient.setListener(GeoPointMapActivity.this);
+            locationClient.setListener(this);
```

Comment:
`this` is the callback, right? Did you try this change?

I think the whole reason we want to update `pmd` is so that we can use `GeoPointMapActivity.this` without getting a pmd error.

Reply:
Then we'll have to exclude that check again

Reply:
> this is the callback, right? Did you try this change?
Yes

> I think the whole reason we want to update pmd is so that we can use GeoPointMapActivity.this without getting a pmd error.

Since `GeoPointMapActivity.this` can be replaced with `this`, so I think the pmd error is correct as the ClassReference is redundant

Reply:
🤔 Well, evidently my original trial was faulty in some way because I really thought the pmd update made a difference but I just tried it again and you're right that it's still flagged.

But here's the key that I had forgotten: Java lambdas are lexically scoped so they don't introduce a new scoping level. https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html#accessing-local-variables has an example illustrating this. So yes, PMD is correct, thanks @shobhitagarwal1612. We can leave the unqualified `this`. CC @jd-alexander

Reply:
Thanks for the updates @shobhitagarwal1612 @lognaturel I know what was causing the confusion. When using lambdas everything works fine. I was getting the error when utilizing the expanded callback like so 
``` 
((SupportMapFragment) getSupportFragmentManager().findFragmentById(R.id.map)).getMapAsync(new OnMapReadyCallback() {
            @Override
            public void onMapReady(GoogleMap googleMap) {
                locationClient.setListener(this);
            }
        });
```
and in this case `this` won't work since it's going to reference the `OnMapReadyCallback` itself. So yes, all I had to do was convert the callback to a lambda to avoid the pmd issues haha.

Reply:
😄 Exactly, I got tripped up by the same -- it kind of makes sense to expect the two variants to have the same semantics!

Reply:
Ohh, and I do think I tried changing the PMD version when it was still an expanded callback. It all makes sense now.